binaries/cuprated/src/blockchain/interface.rs

```diff
@@ -0,0 +1,168 @@
+//! The blockchain manger interface.
+//!
+//! This module contains all the functions to mutate the blockchain's state in any way, through the
+//! blockchain manger.
```

Comment:
This file contains the interface for the blockchain manager.

I decided to not go with a tower::service interface, as I just don't think it is needed here. The benefits of a Service interface here would pretty much be uniformity with every other service, the drawbacks are the extra over head of holding/cloning another handle (admittedly low) and just the extra hassle of holding another service handle.

I think these free functions as a way to mutate the blockchain makes more sense.

Reply:
That's fair, the API here should be pretty easy to use - all you need to do is call the functions, I'll add some more docs.

> Couldn't you say this about every Service though

Other service are exposed this one is not. There are other benefits to the Service interface, like the wide range of middleware but since we are not going to make use of them for the blockchain manager that isn't a benefit here.